lib/icingadb/redisconnection.hpp

```diff
@@ -387,6 +394,67 @@ void RedisConnection::WriteOne(StreamPtr& stream, RedisConnection::Query& query,
 	}
 }
 
+/**
+ * Initialize a Redis stream
+ *
+ * @param stream Redis server connection
+ * @param query Redis query
+ */
+template<class StreamPtr>
+void RedisConnection::Handshake(StreamPtr& stream, boost::asio::yield_context& yc)
+{
+	auto strm (stream);
+
+	if (m_Password.IsEmpty() && !m_DbIndex) {
+		// Trigger NOAUTH
+		WriteRESP(*strm, {"PING"}, yc);
+	} else {
+		if (!m_Password.IsEmpty()) {
+			WriteRESP(*strm, {"AUTH", m_Password}, yc);
```

Comment:
Starting with Redis 6, it's also possible to (optionally) pass a username: https://redis.io/commands/auth

Do we also want to support this?

Reply:
Maybe in 2.14.